src/saml.ts

```diff
@@ -352,6 +352,10 @@ class SAML {
       request["samlp:AuthnRequest"]["samlp:Scoping"] = scoping;
     }
 
+    if (this.options.samlExtensions != null) {
+      request["samlp:AuthnRequest"]["samlp:Extensions"] = this.options.samlExtensions;
```

Comment:
Position of `samlp:Extensions` is defined in SAML protocol schema.
```xml
…
    <complexType name="RequestAbstractType" abstract="true">                                                 
        <sequence>                                                                                           
            <element ref="saml:Issuer" minOccurs="0"/>                                                       
            <element ref="ds:Signature" minOccurs="0"/>                                                      
            <element ref="samlp:Extensions" minOccurs="0"/>                                                  
        </sequence>                                                                                          
        <attribute name="ID" type="ID" use="required"/>                                                      
        <attribute name="Version" type="string" use="required"/>                                             
        <attribute name="IssueInstant" type="dateTime" use="required"/>                                      
        <attribute name="Destination" type="anyURI" use="optional"/>                                         
        <attribute name="Consent" type="anyURI" use="optional"/>                                             
    </complexType>
…
    <complexType name="AuthnRequestType">
        <complexContent>
            <extension base="samlp:RequestAbstractType">
                <sequence>
                    <element ref="saml:Subject" minOccurs="0"/>
                    <element ref="samlp:NameIDPolicy" minOccurs="0"/>
                    <element ref="saml:Conditions" minOccurs="0"/>
                    <element ref="samlp:RequestedAuthnContext" minOccurs="0"/>
                    <element ref="samlp:Scoping" minOccurs="0"/>
                </sequence>
                <attribute name="ForceAuthn" type="boolean" use="optional"/>
                <attribute name="IsPassive" type="boolean" use="optional"/>
                <attribute name="ProtocolBinding" type="anyURI" use="optional"/>
                <attribute name="AssertionConsumerServiceIndex" type="unsignedShort" use="optional"/>
                <attribute name="AssertionConsumerServiceURL" type="anyURI" use="optional"/>
                <attribute name="AttributeConsumingServiceIndex" type="unsignedShort" use="optional"/>
                <attribute name="ProviderName" type="string" use="optional"/>
            </extension>
        </complexContent>
    </complexType>
…
```
source: https://docs.oasis-open.org/security/saml/v2.0/saml-schema-protocol-2.0.xsd


It looks like this implementation would place it into wrong place if developer has configured also e.g. `Scoping`

Reply:
thanks mate, you are right. it is at wrong place. I'll fix it.